pallets/did/src/lib.rs

```diff
@@ -589,15 +600,18 @@ pub mod pallet {
 			Did::<T>::insert(&did_identifier, did_entry.clone());
 
 			// Withdraw the fee. We made sure that enough balance is available. But if this
-			// fails, we don't withdraw anything.
-			let imbalance = <T::Currency as Currency<AccountIdOf<T>>>::withdraw(
+			// fails, we don't withdraw anything. Since we do the best effort we can unwrap
+			// the result.
+			let imbalance: CreditOf<T> = <T::Currency as Balanced<AccountIdOf<T>>>::withdraw(
 				&did_entry.deposit.owner,
 				T::Fee::get(),
-				WithdrawReasons::FEE,
-				ExistenceRequirement::AllowDeath,
+				Precision::BestEffort,
+				Preservation::Protect,
+				Fortitude::Polite,
 			)
-			.unwrap_or_else(|_| NegativeImbalanceOf::<T>::zero());
+			.unwrap();
```

Comment:
I do not know, how I could miss that: [45a2be7](https://github.com/KILTprotocol/kilt-node/pull/527/commits/45a2be7b50bdc4191ee91b805d2344961d33a202)

Reply:
Why not keeping the `unwrap_or_else` logic as before?

Reply:
I think in this case it does not really do any difference. Since we're using `BestEffort`. In the worst case, imbalance is zero. Using `unwrap_or_else` could be misleading, since we should never throw an error.

Reply:
> Ok, but we still throw an error, right?  
In general, no, we do not. We would only throw an error If we would use `Exact` as precision. From the Docs:  If `precision` is `BestEffort` and `value` in full could not be removed (e.g. due to underflow), then the maximum is removed. 

>  If the user does not have enough funds to pay for a DID creation fee, the operation should fail altogether
I changed the parameters and removed the comments: [f9b4fb7](https://github.com/KILTprotocol/kilt-node/pull/527/commits/f9b4fb7439008968045f2af4f0b582e915a31494)

Reply:
Yeah the API exposed by Substrate is not the best, as it is normal, in Rust, to assume a function can fail if it returns an `Err` 😁 Anyway, I think the PR you linked has to do with the staking pallet and not the DID, but it anyway looks good now.